install-ambari-python.sh

```diff
@@ -0,0 +1,188 @@
+  fi
+  if [ -d "$SCRIPT_DIR/ambari_python.egg-info" ]; then
+    echo "Removing '$SCRIPT_DIR/ambari_python.egg-info' directoy ..."
+    rm -r "$SCRIPT_DIR/ambari_python.egg-info"
+    echo "Directory '$SCRIPT_DIR/ambari_python.egg-info' successfully deleted."
+  fi
+  if [ -d "$SCRIPT_DIR/target/ambari-python-dist" ]; then
+    echo "Removing '$SCRIPT_DIR/target/ambari-python' directoy ..."
+    rm -r "$SCRIPT_DIR/target/ambari-python-dist"
+    echo "Directory '$SCRIPT_DIR/target/ambari-python' successfully deleted."
+  fi
+}
+
+function generate_site_packages() {
+  local version="$1"
+  pip install $SCRIPT_DIR/dist/ambari-python-$version.tar.gz -I --install-option="--prefix=$SCRIPT_DIR/target/ambari-python-dist"
+}
+
+function archive_python_dist() {
+  local artifact="$1"
+  local site_packages_dir=$(find $SCRIPT_DIR/target/ambari-python-dist -name "site-packages")
+  local base_dir="`dirname $site_packages_dir`" # use this to make it work with different python versions
+  if [[ -f "$SCRIPT_DIR/target/$artifact" ]]; then
+    echo "Removing '$SCRIPT_DIR/target/$artifact' file ..."
+    echo "File '$SCRIPT_DIR/target/$artifact' successfully deleted."
+  fi
+  tar -zcf $SCRIPT_DIR/target/$artifact -C $base_dir site-packages
+}
+
+function install() {
+  local artifact_file="$1"
+  local version="$2"
+  mvn install:install-file -Dfile=$artifact_file -DgeneratePom=true -Dversion=$version -DartifactId=ambari-python -DgroupId=org.apache.ambari -Dpackaging=tar.gz
+}
+
+function deploy() {
+  local artifact_file="$1"
+  local version="$2"
+  local repo_id="$3"
+  local repo_url="$4"
+  mvn deploy:deploy-file -Dfile=$artifact_file -Dpackaging=tar.gz -DgeneratePom=true -Dversion=$version -DartifactId=ambari-python -DgroupId=org.apache.ambari -Durl="$repo_url" -DrepositoryId="$repo_url"
+}
+
+function run_setup_py() {
+  local version="$1"
+  if [[ ! -z "$version" ]]; then
+    env AMBARI_VERSION="$version" python setup.py sdist
+  else
+    python setup.py sdist
+  fi
+}
+
+function main() {
+  while [[ $# -gt 0 ]]
+    do
+      key="$1"
+      case $key in
+        -d|--deploy)
+          local DEPLOY="true"
+          shift 1
+        ;;
+        -c|--clean)
+          local CLEAN="true"
+          shift 1
+        ;;
+        -v|--version)
+          local VERSION="$2"
+          shift 2
+        ;;
+        -i|--repository-id)
+          local REPOSITORY_ID="$2"
+          shift 2
+        ;;
+        -r|--repository-url)
+          local REPOSITORY_URL="$2"
+          shift 2
+        ;;
+        -h|--help)
+          shift 1
+          print_help
+          exit 0
+        ;;
+        *)
+          echo "Unknown option: $1"
+          exit 1
+        ;;
+      esac
+  done
+
+  if [[ -z "$DEPLOY" ]] ; then
+    DEPLOY="false"
+  fi
+
+  if [ "$(uname -s)" = 'Linux' ]; then
+    SCRIPT_DIR="`dirname "$(readlink -f "$0")"`"
+  else
+    SCRIPT_DIR="`dirname "$(readlinkf "$0")"`"
+  fi
+
+  clean
+  if [[ "$CLEAN" == "true" ]]; then
+    return 0
```

Comment:
Always "cleans" (with or without `-c`), but exits without further action with `-c`.  Is it intentional?

Reply:
yes. if clean is used as a flag, it stops there, otherwise before regeneration we are deleting the dist folders

Reply:
Odd.